main.py

```
@@ -78,6 +86,19 @@
                 logger.info(
                     f"Successfully retrieved the website content (status code: {response.status_code})."
                 )
+
+                end_time = time.time()
+                execution_time = end_time - start_time
+                memory_usage = tracemalloc.get_traced_memory()
+                cpu_usage = process.cpu_percent(interval=None)
+                network_latency = response.elapsed.total_seconds()
+
+                logger.info(f"Execution time for scraping {url}: {execution_time:.2f} seconds")
```

Ruff (0.8.2)

Logging statement uses f-string

(G004)